External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -1982,11 +1982,11 @@ public static int GetStartOfStatement(ScintillaControl sci, int statementEnd, AS
                 else if (c == '[' && arrCount > 0) arrCount--;
                 else if (c == ')') parCount++;
                 else if (c == '(' && parCount > 0) parCount--;
-                else if (c == '>')
+                else if (c == '>' && arrCount == 0 && parCount == 0 && braCount == 0)
```

Comment:
I'm really curious if this is 100% fine, as there are many cases were generics may appear between []{}(), but I cannot check at the moment, also it seems there are no tests with this?. For example `arr = [new Array<String>()]`, `function test(type:Class<Dynamic>)`, etc.

Reply:
5 min

Reply:
All for you ^__^

Reply:
generics is not a problem, problem is operators `<`, `>`

Reply:
>For example arr = [new Array<String>()], function test(type:Class<Dynamic>), etc.

2918: `new TestCaseData(" new Vector.<int>()$(EntryPoint)"`
2927: `new TestCaseData(" new <Vector.<Object>>[new <Object>[{}]]$(EntryPoint)"`
2970: `new TestCaseData(" new Array<{name:String, params:Array<Dynamic>}>()$(EntryPoint)"`

Reply:
I see, but in Haxe "=>" is an operator and could appear in other places to be later processed at macro time, so "=>" should be processed specifically and not assume it's going to be between brackets.

Reply:
in the next PRs there will be a lot of them